### PR TITLE
Request handler test

### PR DESCRIPTION
**Pull request by kfrankc**
Unit testing for RequestHandler class. Also modify Makefile to accomodate for compiling the `request_handlers_test.cc` file.

### CODE REVIEW

**Comment by KyCodeHuynh** [comment]
What necessitated this last commit: https://github.com/UCLA-CS130/Mr.-Robot-et-al./pull/26/commits/c0f458afb0e618ab98ee0c3c4ecf4f347014c391?

File: Makefile [hunk 1]
```diff
@@ -35,9 +35,12 @@ $(TESTS):
 	# Build our own tests, using GTest
 	# TODO: Figure out how to name multiple CC and produce corresponding object for each
 	$(CXX) $(SRC_FLAGS) $(GTEST_FLAGS) $(PARSER_PATH)config_parser.cc server_config.cc $(TESTS).cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(LDFLAGS) -o $(TESTS)
+	# $(CXX) $(SRC_FLAGS) $(GTEST_FLAGS) $(TESTS).cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a $(LDFLAGS) -o $(TESTS)
```
Comment (apfong) [inline Makefile]: Don't think we need this line here, it's pretty much identical to the one above except without `$(PARSER_PATH)config_parser.cc` and `server_config.cc`

Comment (KyCodeHuynh) [inline Makefile]: I've added test building de-duplication as a task for Issue #11

File: request_handlers_test.cc [hunk 2]
```diff
@@ -0,0 +1,111 @@
+#include "request_handlers.h"
+#include "gtest/gtest.h"
+
+#include <iostream>
```
Comment (KyCodeHuynh) [inline request_handlers_test.cc]: It doesn't look like `iostream` is needed below, as no debug output remains.

File: request_handlers_test.cc [hunk 3]
```diff
@@ -0,0 +1,111 @@
+#include "request_handlers.h"
+#include "gtest/gtest.h"
```
Comment (KyCodeHuynh) [inline request_handlers_test.cc]: Can we reorder these includes to be alphabetical? I totally agree that the gtest header is less important, but this just maintains a consistent style.

File: request_handlers_test.cc [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+#include <iostream>
+
+using namespace lightning_server;
+
+const int MAX_REQ_SIZE = 8192; // bytes
+
+class RequestHandlersTest : public ::testing::Test {
+protected:
+  // Helper method to pass arbitrary strings into the parser
```
Comment (KyCodeHuynh) [inline request_handlers_test.cc]: It looks like this comment was accidentally left-over from the nginx-configparser tests?

File: request_handlers_test.cc [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  bool HandleRequest(char* request_buffer,
+                     char* response_buffer,
+                     size_t& response_buffer_size) {
+    request_handlers::echoRequestHandler(request_buffer,
+                                         response_buffer,
+                                         response_buffer_size);
+    if (response_buffer == nullptr) {
+      return false;
+    } else {
```
Comment (KyCodeHuynh) [inline request_handlers_test.cc]: Our current style for if-else has `else` on the next line: 

```c++
if (...) {
  ...
else {
  ...
}
```

File: request_handlers_test.cc [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  char request_buffer[MAX_REQ_SIZE];
+
+  // Customize message in request buffer
+  strcpy(request_buffer, "");
+
+  // Handle echo response in external handler
+  char response_buffer[MAX_REQ_SIZE];
+  size_t response_size = 0;
+
+  EXPECT_TRUE(HandleRequest(request_buffer, response_buffer, response_size))
```
Comment (apfong) [inline request_handlers_test.cc]: Could this still pass if HandleRequest set response_buffer to anything that isn't nullptr, like `'sadl;fkjakd'`?

File: request_handlers_test.cc [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  char request_buffer[MAX_REQ_SIZE];
+
+  // Customize message in request buffer
+  strcpy(request_buffer, "abcd\n");
+
+  // Handle echo response in external handler
+  char response_buffer[MAX_REQ_SIZE];
+  size_t response_size = 5;
+
+  EXPECT_TRUE(HandleRequest(request_buffer, response_buffer, response_size))
```
Comment (apfong) [inline request_handlers_test.cc]: Is this test only expecting response_buffer to be something other than `nullptr`? And if so, it seems like it's not really checking if the handler responds correctly to a random string, only that it isnt returning nullptr.

**Pull request merged**